### PR TITLE
chore: turn RunPhase in a class with helper methods

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: packages/playwright-test/src/runner.ts [hunk 1]
```diff
@@ -251,14 +273,13 @@ export class Runner {
         throw new Error(`Cannot find project group '${projectGroup}' in the config`);
       for (const entry of group) {
```
Comment (dgozman) [inline packages/playwright-test/src/runner.ts:274]: This whole method could move to `RunPhase`?

File: packages/playwright-test/src/runner.ts [hunk 2]
```diff
@@ -424,6 +444,15 @@ export class Runner {
       const testGroups = createTestGroups(projectSuites, config.workers);
       concurrentTestGroups.push(testGroups);
     }
+    return { rootSuite, concurrentTestGroups };
+  }
+
+  private async _run(options: RunOptions): Promise<FullResult> {
```
Comment (dgozman) [inline packages/playwright-test/src/runner.ts:450]: Why this method split? Now `7. Fail when no tests` seems strange :)

Comment (yury-s) [reply packages/playwright-test/src/runner.ts:450]: For better readability. Removed numbers.

File: packages/playwright-test/src/runner.ts (hunk 1 above)
Comment (yury-s) [reply packages/playwright-test/src/runner.ts:274]: Done.

**Review by dgozman — APPROVED** [review]

**Pull request merged**